python-package/lightgbm/sklearn.py

```diff
@@ -126,102 +129,115 @@ def inner(preds, dataset):
             Maximum tree leaves for base learners.
-        max_depth : int
+        max_depth : int, optional (default=-1)
             Maximum tree depth for base learners, -1 means no limit.
-        learning_rate : float
+        learning_rate : float, optional (default=0.1)
             Boosting learning rate.
-        n_estimators : int
+        n_estimators : int, optional (default=10)
             Number of boosted trees to fit.
-        max_bin : int
+        max_bin : int, optional (default=255)
             Number of bucketed bin for feature values.
-        subsample_for_bin : int
+        subsample_for_bin : int, optional (default=50000)
             Number of samples for constructing bins.
-        objective : string or callable
+        objective : string, callable or None, optional (default=None)
             Specify the learning task and the corresponding learning objective or
             a custom objective function to be used (see note below).
-            default: binary for LGBMClassifier, lambdarank for LGBMRanker.
-        min_split_gain : float
+            default: 'binary' for LGBMClassifier, 'lambdarank' for LGBMRanker.
+        min_split_gain : float, optional (default=0.)
             Minimum loss reduction required to make a further partition on a leaf node of the tree.
-        min_child_weight : int
+        min_child_weight : int, optional (default=5)
             Minimum sum of instance weight(hessian) needed in a child(leaf).
-        min_child_samples : int
+        min_child_samples : int, optional (default=10)
             Minimum number of data need in a child(leaf).
-        subsample : float
+        subsample : float, optional (default=1.)
             Subsample ratio of the training instance.
-        subsample_freq : int
-            frequence of subsample, <=0 means no enable.
-        colsample_bytree : float
+        subsample_freq : int, optional (default=1)
+            Frequence of subsample, <=0 means no enable.
+        colsample_bytree : float, optional (default=1.)
             Subsample ratio of columns when constructing each tree.
-        reg_alpha : float
+        reg_alpha : float, optional (default=0.)
             L1 regularization term on weights.
-        reg_lambda : float
+        reg_lambda : float, optional (default=0.)
             L2 regularization term on weights.
-        random_state : int
+        random_state : int, optional (default=0)
             Random number seed.
-        n_jobs : int
+        n_jobs : int, optional (default=-1)
             Number of parallel threads.
-        silent : boolean
+        silent : bool, optional (default=True)
             Whether to print messages while running boosting.
         **kwargs : other parameters
             Check http://lightgbm.readthedocs.io/en/latest/Parameters.html for more parameters.
             Note: **kwargs is not supported in sklearn, it may cause unexpected issues.
```

Comment:
Is this note still actual?

Reply:
I can't ensure everything works because sklearn doesn't support **kwargs and it indeed raises some issues before.

Reply:
@wxchan OK, got it!